network/src/noise/handshake.rs

```diff
@@ -179,11 +187,11 @@ impl NoiseUpgrader {
     /// Perform an outbound protocol upgrade on this connection.
     ///
     /// This runs the "client" side of the Noise IK handshake to establish a
-    /// secure Noise stream and exchange static public keys. In mutual auth
-    /// scenarios, we will also include an anti replay attack counter in the
+    /// secure Noise stream and send its static public key to the server.
+    /// In mutual auth scenarios, we will also include an anti replay attack counter in the
     /// Noise handshake payload. Currently this counter is always a millisecond-
     /// granularity unix epoch timestamp.
-    pub async fn upgrade_outbound<TSocket>(
+    pub async fn upgrade_outbound_with_noise<TSocket>(
```

Comment:
why rename these? should be clear from the namespace...

Reply:
reasons:

- we kept confusing these in discussions
- for the specification it's not super clean to have name collisions

in general I think it can be confusing in an implementation as well, but I don't have a strong enough opinion to fight it if you think we should rename it back :p

Reply:
we can distinguish with `noise::upgrade_inbound` vs `transport::upgrade_inbound`? the reason why i mention this is because I think we can (in the future) extract this sort of "upgrade" structure into a trait for upgrading existing sockets (different from dial/listen which produce a new socket/stream of sockets). not sure what this would look like, but maybe

```rust
trait Upgrade {
    type Output;

    fn parse_outbound_addr(protos: &[Protocol]) -> Option<(/* addr args? */, &[Protocol])>;

    async fn upgrade_inbound(/* socket? addr args? */) -> Result<Self::Output>;

    async fn upgrade_outbound(/* socket? addr args? */) -> Result<Self::Output>;
}
```

and then NoiseUpgrader would impl `Upgrade` for example

Reply:
ait, renamed back